frontend/src/core/loaders/components/WaveLoader.js

```diff
@@ -1,43 +1,45 @@
 /* @flow */
 
 import React from 'react';
-import { connect } from 'react-redux';
 
 import './WaveLoader.css';
 
-import type { LocaleState } from 'modules/core/locales/reducer';
-import type { L10nState } from 'modules/core/l10n/reducer';
-
 type Props = {
     isLoading: boolean,
     children?: React.Node,
 };
 
-const WaveLoadingBar = (): React.Node => (
-    <div id="project-load" className="overlay">
-        <div className="inner">
-            <div className="animation">
-                <div></div>
-                &nbsp;
-                <div></div>
-                &nbsp;
-                <div></div>
-                &nbsp;
-                <div></div>
-                &nbsp;
-                <div></div>
+export class WaveLoadingBar extends React.Component {
+    render() {
+        return (<div id="project-load" className="overlay">
+            <div className="inner">
+                <div className="animation">
+                    <div></div>
+                    &nbsp;
+                    <div></div>
+                    &nbsp;
+                    <div></div>
+                    &nbsp;
+                    <div></div>
+                    &nbsp;
+                    <div></div>
+                </div>
+                <div className="text">&quot;640K ought to be enough for anybody.&quot;</div>
             </div>
-            <div className="text">&quot;640K ought to be enough for anybody.&quot;</div>
-        </div>
-    </div>
-);
+        </div>);
+    }
+}
 
-const WaveLoader = (props: Props): React.Node => (
+// const WaveLoader = (props: Props): React.Node => React(
+//     // Render children components, because some of fetches happen after components are rendered.
+//      props.isLoading ? <WaveLoadingBar /> : props.children
+// );
+class WaveLoader extends React.Component<Props> {
+    render() {
+    const props = this.props;
     // Render children components, because some of fetches happen after components are rendered.
-    <React.Fragment>
-        { props.isLoading ? <WaveLoadingBar /> : '' }
-        { props.children }
-    </React.Fragment>
-);
+        return props.isLoading ? <WaveLoadingBar /> : props.children;
+    }
+}
```

Comment:
This is great, but I would go even further, remove this component, and put the logic in the `App` component. I don't think there's much use in having a component for just this `if`.

Reply:
done :-)